### PR TITLE
Clarify gen data docs

### PR DESCRIPTION
**Pull request by yngve-sk**
Goal: Describe less ambiguously what a GEN_DATA result file must contain, why it is specified and how it is used. Also remove deprecated `INPUT_FORMAT` doc

### CODE REVIEW

**Review by jonathan-eq — APPROVED** [review]
Great improvement! It was more clearly defined now.

File: docs/reference/configuration/keywords.rst [hunk 1]
```diff
@@ -790,16 +790,21 @@ and/or history matching project.
 .. _gen_data:
 .. topic:: GEN_DATA
 
-        The ``GEN_DATA`` keyword is used to load text files which have been generated
-        by the forward model.
+        The ``GEN_DATA`` keyword is used to declare text files that are referred to by
+        :ref:`GENERAL_OBSERVATIONs <general_observation>`. It declares
+        text files which have been generated by the forward model, which will be
+        loaded by ert when loading general observations.
+        These text files naming scheme may be arbitrary. The contents of these result files are always:
```
[the remark this answers is not in the document]
Comment (yngve-sk) [reply docs/reference/configuration/keywords.rst:797]: Fixing

File: docs/reference/configuration/keywords.rst [hunk 2]
```diff
@@ -790,16 +790,24 @@ and/or history matching project.
 .. _gen_data:
 .. topic:: GEN_DATA
 
-        The ``GEN_DATA`` keyword is used to load text files which have been generated
-        by the forward model.
+        The ``GEN_DATA`` keyword is used to declare text files that are referred to by
```
Comment (oyvindeide) [inline docs/reference/configuration/keywords.rst:793]: I would rewrite this a little, for example that: 
```
The GEN_DATA key is used to declare a response which corresponds to a GENERAL_OBSERVATION. It expects to read a text file produced by the forward model, which will be loaded...

File: docs/reference/configuration/keywords.rst [hunk 3]
```diff
@@ -790,16 +790,24 @@ and/or history matching project.
...
+        :ref:`GENERAL_OBSERVATIONs <general_observation>`. It declares
+        text files which have been generated by the forward model, which will be
+        loaded by ert when loading general observations.
+        These text files are expected to follow the same naming scheme for all realizations
+        (ex: ``gd_%d`` which may resolve to ``gd_0``, ``gd_1`` where `%d` is the report step).
+        The contents of these result are always of this format:
+        **exactly one floating point number per line**.
+        If we speak about "indexing" into a ``GEN_DATA`` file,
```
Comment (oyvindeide) [inline docs/reference/configuration/keywords.rst:801]: Would reword this slightly, perhaps replacing: `If we speak about "indexing"` ->` Indexing ``GEN_DATA`` refers to the row number...`

**Review by oyvindeide — COMMENTED** [review]
Thanks for updating the documentation, always welcomed to see! I just have some minor comments.

**Review by oyvindeide — APPROVED** [review]
LGTM!
